src/test/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpConnectionStateTest.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2016-2020 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.amqp.internal.stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.CLOSE_PIPE;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.CLOSE_RCVD;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.CLOSE_SENT;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.DISCARDING;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.END;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.HDR_EXCH;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.HDR_RCVD;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.HDR_SENT;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.OC_PIPE;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.OPENED;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.OPEN_PIPE;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.OPEN_RCVD;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.OPEN_SENT;
+import static org.reaktivity.nukleus.amqp.internal.stream.AmqpConnectionState.START;
+
+import org.junit.Test;
+
+public class AmqpConnectionStateTest
+{
+    @Test
+    public void shouldTransitionFromStartToHeaderReceived() throws Exception
+    {
+        AmqpConnectionState state = START.receivedHeader();
+        assertEquals(HDR_RCVD, state);
+    }
```

Comment:
Suggest you have one test method for each state, called something like `shouldTransitionFromStart()`  and then have assert for all the possible transitions, good and bad.

For example:
```
    @Test
    public void shouldTransitionFromStart() throws Exception
    {
        assertEquals(HDR_SENT, START.sentHeader());
        assertEquals(HDR_RCVD, START.receivedHeader());
        assertEquals(ERROR, START.sentOpen());
        assertEquals(ERROR, START.receivedOpen());
        assertEquals(ERROR, START.sentClose());
        assertEquals(ERROR, START.receivedClose());
    }
```